neo/rawio/axonarawio.py

```diff
@@ -262,13 +261,17 @@ def _segment_t_stop(self, block_index, seg_index):
                 raise ValueError('Can not determine common tetrode recording'
                                  'duration.')
 
-            tetrode_duration = int(self.file_parameters['unit']['duration'])
+            tetrode_duration = float(self.file_parameters['unit']['duration'])
             t_stop = max(t_stop, tetrode_duration)
 
         return t_stop
 
     def _get_signal_size(self, block_index, seg_index, channel_indexes=None):
-        return self.file_parameters['bin']['num_total_samples']
+        if 'num_total_packets' in self.file_parameters['bin']:
+            return self.file_parameters['bin']['num_total_samples']
+        else:
+            sr = self.file_parameters['set']['sampling_rate']
+            return int(float(self.file_parameters['unit']['duration']) * sr)
```

Comment:
I don't think units should be taken into account here as `signal` in this context refers to continuous signals. It is safer to just return 0 when no signal is present.
```suggestion
        else:
            return 0
```